Support Takazine's SabreBerry32 DAC Card

SatoruKawase opened the pull request:
I have created audiophile-grade DAC card driver and ESS SABRE9018Q2C dac chip driver that DAC card use ESS SABRE9018Q2C chip.
SabreBerry32 DAC card is yet to supoort international users. Takazine plan to distribute international via SWITCH SCIENCE like SabreBerry Plus DAC card (https://international.switch-science.com/catalog/2118/).

SabreBerry32 DAC use DAC side I2S master mode (Raspberry Pi side I2S I/F slave).

SabreBerry32 DAC card support page (only Japanese yet)
http://nw-electric.way-nifty.com/blog/sabreberry32.html

SatoruKawase committed:
Support Takazine's SabreBerry32 DAC Card

clivem commented:
@SatoruKawase I'd be very interested in adding one of these boards to my Pi I2S "test farm". Where can I purchase it from? (It's not available for sale on that Switch Science web site. Or if it is, I can't find it.)

pelwell commented:
We have previously accept Pi-specific drivers - the sound card driver is a good example - but we normally expect more generic modules like the codec driver to be submitted upstream.

What do you think, @popcornmix?

If you were to send it upstream then the maintainers would probably expect the driver to be called es9108.c. I would also like it if the Makefile and README additions were inserted in alphabetical order.

SatoruKawase commented:
@clivem SabreBerry32 yet available for international. Please tell me your address, Takazine (Mr. Nishiara) will ship a prototype DAC board to you. The difference prototype with production is PWB pattern which has through hole for the capacitor to use decoupling of SABRE9018Q2C internal headphone amplifier's -3.3V power supply.

clivem commented:
> If you were to send it upstream then the maintainers would probably expect the driver to be called es9108.c

@pelwell Not sure I agree with that, Phil. All the ESS DAC chips used to have "ES" prefix, but they differentiate the "mobile" parts now by prefix, (ie. low power and built-in headphone amp), with "SABRE" prefix. I would agree with OP that codec file should be named, sabre9018q2c.c.

The part that I am uncomfortable with, is the naming of the product, Sabre**Berry**32. I suppose HiFiBerry should be flattered, in the same way that PiF must be, every time there is the launch of a new fruit Pi. ;)

lurch commented on sound/soc/bcm/Kconfig:
```diff
@@ -47,6 +47,13 @@ config SND_BCM2708_SOC_HIFIBERRY_AMP
+        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
+        select SND_SOC_SABRE9018Q2C
+        help
+         Say Y or M if you want to add support for HifiBerry DAC.
```
I guess this needs correcting?

SatoruKawase replied:
Thank you for reviewing. I corrected the typo.

SatoruKawase committed:
Update Kconfig

That is typo.  I fixed help message.

SatoruKawase committed:
DAC Master/Slave mode configure by device tree parameter.

SatoruKawase commented:
I added Master/Slave Mode select function via a dtoverlay param.

For example,
- Master Mode : dtoverlay=sabreberry32-dac
- Slave Mode : dtoverlay=sabreberry32-dac,slave

pelwell commented:
Please don't include the "-overlay" part in the dtoverlay directive. This won't work on the latest rpi-4.4.y builds.

SatoruKawase committed:
Set the module owner on the machine driver.
When slave mode, DAI name and stream_name indicate "SLAVE" mode.
Change flag instance name of dts.

SatoruKawase committed:
Fix ASRC slow lock problem. (Master & Slave Mode)

SatoruKawase committed:
Change volume control name.
Change machine driver name.
Change copyright message.
Change source format TAB is 8 white spaces & 80 columns.

SatoruKawase committed:
Add I2S S16 format on DAC slave mode.

FishTest commented:
are there any diffent between es9018k2m and es9018q2c in drivers?
if possible,could you make an es9018k2m version driver?
